app/V1Module/security/Policies/ShadowAssignmentEvaluationPermissionPolicy.php

```diff
@@ -0,0 +1,33 @@
+<?php
+namespace App\Security\Policies;
+
+use App\Model\Entity\ShadowAssignmentEvaluation;
+use App\Security\Identity;
+
+class ShadowAssignmentEvaluationPermissionPolicy implements IPermissionPolicy {
+  public function getAssociatedClass() {
+    return ShadowAssignmentEvaluation::class;
+  }
+
+  public function isSupervisor(Identity $identity, ShadowAssignmentEvaluation $evaluation) {
+    $assignment = $evaluation->getShadowAssignment();
+    $group = $assignment->getGroup();
+    $user = $identity->getUserData();
+
+    if ($user === null) {
+      return false;
+    }
+
+    return $group->isSupervisorOf($user) || $group->isAdminOf($user);
+  }
+
+  public function isEvaluatee(Identity $identity, ShadowAssignmentEvaluation $evaluation) {
```

Comment:
Wow, a new word!

Reply:
https://en.wiktionary.org/wiki/evaluatee

Reply:
I know, I just didn't know it existed.